state-sync/src/coordinator.rs

```diff
@@ -609,7 +666,8 @@ impl<T: ExecutorProxyTrait> StateSyncCoordinator<T> {
             TargetType::Waypoint(waypoint_version) => {
                 self.process_request_waypoint(peer, request, waypoint_version)
             }
-        }
+        };
+        Ok(result?)
```

Comment:
If you can do a `?` will the `result` not suffice?  like just drop the semicolon?

Reply:
not sure this case, but `?` can do the implicit .into conversion but plain value can't

Reply:
Yeah, we need to convert the anyhow error into an error::Error. We could also do a map_err(), but this is shorter.